components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -929,36 +873,12 @@ subroutine shoc_tend_e3sm( &
        cloud_frac(i,k) = min(cloud_frac(i,k),1._r8)
       enddo
    enddo
-       
-   !obtain water vapor mmr which is a "dry" mmr at this point from the SHOC output
-   qv_dry(:,:) = edsclr_in(:,:,ixq)
-   !----------------
-   !DRY-TO-WET MMRs:
-   !----------------
-   !Since the host model needs wet mixing ratio tendencies(state vector has wet mixing ratios),
-   !we need to convert dry mixing ratios from SHOC to wet mixing ratios before extracting tendencies
-   !NOTE:Function calculate_wetmmr_from_drymmr takes 2 arguments: (wet mmr and "dry" water vapor
-   !mixing ratio)
-       do ixind=1,edsclr_dim
-      if(convert_back_to_wet(ixind)) then
-         edsclr_out(:,:,ixind) = calculate_wetmmr_from_drymmr(ncol, pver, edsclr_in(:,:,ixind), qv_dry)
-      else
-         edsclr_out(:,:,ixind) = edsclr_in(:,:,ixind)
-      endif
-       enddo      
-   !convert state1%q to wet mixing ratios
-   qv_dry(:,:) = state1%q(:,:,ixq)
-   icnt = 0
-   do ixind = 1, pcnst
-      if (lq(ixind))  then
-         icnt = icnt + 1
-         if(convert_back_to_wet(icnt)) then !convert from wet to dry mmr if true
-            state1%q(:,:,ixind) = calculate_wetmmr_from_drymmr(ncol, pver, state1%q(:,:,ixind), qv_dry)
-         endif
-      endif
-     enddo
-   rcm(:,:) = calculate_wetmmr_from_drymmr(ncol, pver, rcm, qv_dry)
-   rtm(:,:) = calculate_wetmmr_from_drymmr(ncol, pver, rtm, qv_dry)
+      
```

Comment:
I think the next few lines are dumb (though I don't think they cause any bugs). If you grep for `edsclr_` you will find that `edsclr_in` is a local variable in shoc_intr.F90 which just gets assigned to values from state%q then passed as an inout variable to shoc_main. After shoc_main, the resulting edsclr_in value is just assigned here to `edsclr_out`... which is all `edsclr_out` is used for. Why not just have one variable named `edsclr` which contains both the input and output versions of edsclr? As an aside, edsclr is the name for the subset of state%q (the advected tracers) which are also turbulently diffused. Can we please come up with a better variable name for this!?!